superset/views/geocoding.py

```diff
@@ -163,14 +162,41 @@ def geocode(self) -> Response:
 
         try:
             data = self._geocode(data)
+        # TODO There is no data returned here
         except Exception as e:
             self.logger.exception(f"Exception when geocoding data {e}")
             if not save_on_stop_geocoding:
                 self.stats_logger.incr("geocoding_failed")
                 return json_error_response(e.args[0])
+        if self.geocoder_util.interruptflag:
+            if not save_on_stop_geocoding:
+                return json_success(json.dumps("geocoding interrupted"))
+        if data[0]:
+            if not save_on_stop_geocoding:
+                return json_error_response(json.dumps(data[0]))
         try:
+            # TODO make this generic only for testing
+            schema = "main"
+            data = data[1]
+            if not data[0]:
```

Comment:
I would check the length of data maybe? checking data[0] here is really confusing since we check it a few lines above to tell if it worked or not